atramhasis/views/views.py

```diff
@@ -350,7 +351,7 @@ def create_treeid(parent_tree_id, concept_id):
         if parent_tree_id is None:
             return str(concept_id)
```

Comment:
Shouldn't this one be urlencoded as well?

Reply:
i believe so, probably it is best to also encode the parent_tree_id ?

Reply:
I tested it with also encoding` str(concept_id)` and it does not seem to work. It only works when only the part after the "|" is encoded. I'm not exactly sure why though...

Reply:
I dug into this. The reason it doesn't work with `str(concept_id)` urlencoded is that 
![image](https://user-images.githubusercontent.com/4599667/234837719-25f686de-68ac-43d1-a452-1a73818e3829.png)
this returns an urlencoded ID but not an urlencoded `concept_id`.

And in the tree.jinja2 we match the concept_id (not urlencoded) with the output of this def create_treeid 

```js

        var ids = id.split('|');
        currentIds = ids;
        var childrenArray = root.children;
        for (i = 0; i < (ids.length - 1); ++i) {
          var child = $.grep(childrenArray, function(child){ return child.concept_id == ids[i]; });
```
I am not a 100% sure if the way this works is entirely desired. But that's at least the reason.

Sidenote; I also wonder if urlencoding like this is enough `http%3A//id.bluebirds.org/bird` python's `quote` by default does not encode the `/` so it's not entirely url-safe persé. You must pass `safe=''` or use `quote_plus` (which would also turn spaces into a `+`

Reply:
> i believe so, probably it is best to also encode the parent_tree_id ?

No, that contains the part that has already been processed and thus already contains urlencoded id's.

Reply:
I think the proper solution is:

- Make sure the treeid consists of <urlencoded id>|<urlencoded id> | <urlencoded id>
- Before processing this, split the treeid on | and urldecode every id.

So, something like:

```
 var ids = id.split('|');  // id = output of create_treeid
       ids = ids.map(x => urldecode(x))
        currentIds = ids;
        var childrenArray = root.children;
        for (i = 0; i < (ids.length - 1); ++i) {
          var child = $.grep(childrenArray, function(child){ return child.concept_id == ids[i]; });
```

or when doing the comparison

```
 var ids = id.split('|');  // id = output of create_treeid
        currentIds = ids;
        var childrenArray = root.children;
        for (i = 0; i < (ids.length - 1); ++i) {
          var child = $.grep(childrenArray, function(child){ return child.concept_id == urldecode(ids[i]); });
```


(no idea how to urldecode in JS, just a guess)

Reply:
The suggested solution seems to work. I had to switch to `quote` + `safe='' `because the java script function does not handle the spaces as `+` by default

Reply:
Tested locallly and works fine now!